src/hooks/usePersistSelection.ts

```diff
@@ -0,0 +1,11 @@
+import {useEffect, useState} from 'react';
+
+function usePersistSelection<TKey extends string | number, TValue>(options: Record<TKey, TValue> | undefined, filter: (option: TValue | undefined) => boolean) {
```

Comment:
The hook name is a little confusing since by default all useState values are persistent. Let's rename to ` useAdaptiveSelection` or any that describes the hook in a clearer way